app/src/main/java/org/mozilla/fenix/settings/search/AddSearchEngineFragment.kt

```diff
@@ -159,22 +167,14 @@ class AddSearchEngineFragment : Fragment(R.layout.fragment_add_search_engine),
                         .getString(R.string.search_add_custom_engine_error_cannot_reach, name)
                 }
                 SearchStringValidator.Result.Success -> {
-                    try {
-                        CustomSearchEngineStore.addSearchEngine(
-                            context = requireContext(),
-                            engineName = name,
-                            searchQuery = searchString
-                        )
-                    } catch (engineNameExists: CustomSearchEngineStore.EngineNameAlreadyExists) {
```

Comment:
Do we want to allow duplicate names here now or should we still tell the user that the name is used?

Reply:
This is now possible. Previously the name (instead of the ID) was used to identify the default search engine. Now that we generate a UUID for custom search engines, having duplicate names is no problem. I think adding the complexity for this check is not needed anymore: If the user wants to add search engines with the same name then so be it.